modules/programs/kitty.nix

```diff
@@ -193,15 +193,7 @@ in {
         '')
 
         (optionalString (cfg.theme != null) ''
-          include ${pkgs.kitty-themes}/share/kitty-themes/${
-            let
-              matching = filter (x: x.name == cfg.theme) (builtins.fromJSON
-                (builtins.readFile
-                  "${pkgs.kitty-themes}/share/kitty-themes/themes.json"));
-            in throwIf (length matching == 0)
-            "kitty-themes does not contain a theme named ${cfg.theme}"
-            (head matching).file
-          }
+          include ${pkgs.kitty-themes}/share/kitty-themes/themes/${cfg.theme}.conf
```

Comment:
Unfortunately this won't work and is not compatible with the activation script block.

For example, if you have `programs.kitty.theme = "Space Gray Eighties";` in your HM configuration. Then the activation script block will work OK but this will produce

    include /nix/store/…/themes/Space Gray Eighties.conf

while the actual file is `/nix/store/…/themes/SpaceGray_Eighties.conf`.

Reply:
So, I think you either have to keep the IFD, or change the meaning of `programs.kitty.theme` such that the user is expected to put the file name. I do not want to include the `themes.json` in the HM repository.

Reply:
Perhaps remove the `theme` option and instead add a `themeFile` option like so:

```nix
themeFile = mkOption {
  type = types.nullOr types.str;
  default = null;
  example = "SpaceGray_Eighties";
  description = ''
    Apply a Kitty color theme. This option takes the file name of a theme in
    a `kitty-themes`, without the `.conf` suffix. See
    <https://github.com/kovidgoyal/kitty-themes/tree/master/themes> for a
    list of themes.
  '';
};
```

Then use `mkChangedOptionModule` to convert existing instances of `programs.kitty.theme` using IFD into the new `themeFile` option.